### ISSUE TITLE
error: could not create a builder instance with TLS data loaded from environment

### ISSUE THREAD

**Issue by lexfrei** [docker/setup-buildx-action#105]
#### Steps to reproduce this issue

1. Install k3s in your arm64 cluster
2. Install [actions-runner-controller](actions-runner-controller)
3. Run your workflow there

#### Expected behaviour

> No errors

#### Actual behaviour

> error: could not create a builder instance with TLS data loaded from environment. Please use `docker context create <context-name>` to create a context for current environment and then create a builder instance with `docker buildx create <context-name>`

### Configuration

* Repository URL (if public): https://github.com/lexfrei/vk2tg/tree/6b67c17cf095a7147a40db4023f2121fd1cb1d9a

* Build URL (if public): https://github.com/lexfrei/vk2tg/runs/3424244864?check_suite_focus=true#step:5:96

```yml
name: Main Workflow

on:
  push

jobs:
  lint:
    runs-on: self-hosted
    steps:
      - uses: actions/checkout@v2.3.4
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v2.5.2
        with:
          version: v1.42
      # Waiting for https://github.com/hadolint/hadolint-action/issues/38
      # - name: hadolint
      #   uses: hadolint/hadolint-action@v1.5.0
      #   with:
      #     dockerfile: build/vk2tg/Dockerfile

  build:
    needs: lint
    runs-on: self-hosted
    steps:

    - name: Docker meta
      id: docker_meta
      uses: docker/metadata-action@v3.5.0
      with:
        images: ghcr.io/${{ github.repository }}
        flavor: |
          latest=true
        tags: |
            type=ref,event=branch
            type=ref,event=pr
            type=sha
    - name: Set up QEMU
      uses: docker/setup-qemu-action@v1.2.0

    - name: Checkout
      uses: actions/checkout@v2.3.4

    - name: Set up Docker Buildx
      uses: docker/setup-buildx-action@v1.5.1

  <...>
```

### Logs

[logs_415.zip](https://github.com/docker/setup-buildx-action/files/7048485/logs_415.zip)

**Comment by crazy-max** [comment]
@lexfrei Looks like you're using a self-hosted runner and the default context is from a tls environment but you can't create a builder instance from a tls environment (see https://github.com/docker/buildx/issues/30

You have to create a context for the current environment with `docker context create <context-name>` like the error message suggest.

**Comment by lexfrei** [comment]
For history:
[Fix](https://github.com/mumoshu/actions-runner-controller-ci/commit/e91c8c0f6ca82aa7618010c6d2f417aa46c4a4bf)

**Comment by DmitriyBobrovskiy** [comment]
> For history: [Fix](https://github.com/mumoshu/actions-runner-controller-ci/commit/e91c8c0f6ca82aa7618010c6d2f417aa46c4a4bf)

For the history if repo will be gone:
<img width="369" alt="image" src="https://user-images.githubusercontent.com/33372992/199765703-f1c4d8bc-71ba-4f0a-9070-d5f6104b8d84.png">

**Comment by joaoluiznaufel** [comment]
but this is coming back to version 1. Someome already know how to do a proper fix using v2?

**Comment by joaoluiznaufel** [comment]
ah, found it, you can use the v2 that it will continue work

**Comment by lifeofmoo** [comment]
> > For history: [Fix](https://github.com/mumoshu/actions-runner-controller-ci/commit/e91c8c0f6ca82aa7618010c6d2f417aa46c4a4bf)
> 
> For the history if repo will be gone: <img alt="image" width="369" src="https://user-images.githubusercontent.com/33372992/199765703-f1c4d8bc-71ba-4f0a-9070-d5f6104b8d84.png">

Hello, I used the findings above to resolve the issue for me.

https://github.com/docker/setup-buildx-action/issues/311

Is anyone able to comment/shed light as to WHY this works. Is this is an accepted approach for some workaround?

**Comment by crazy-max** [comment]
> For history: [Fix](https://github.com/mumoshu/actions-runner-controller-ci/commit/e91c8c0f6ca82aa7618010c6d2f417aa46c4a4bf)

Made some changes in #341 to automatically create a docker context is TLS data are loaded for default context if someone is willing to test on his self-hosted environment :pray::

```yaml
      -
        name: Set up Docker Buildx
        uses: crazy-max/docker-setup-buildx-action@docker-context-tls
```

**Comment by crazy-max** [comment]
Should be fixed since [3.6.0](https://github.com/docker/setup-buildx-action/releases/tag/v3.6.0)
